cvat/apps/engine/tests/utils.py

```diff
@@ -125,6 +128,252 @@ def setUp(self):
+
+        rq_id = response.json().get("rq_id")
+        if expected_4xx_status_code:
+            # export task by admin to get real rq_id
+            response = self._post_request(api_path, self.admin)
+            self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+            rq_id = response.json().get("rq_id")
+
+        assert rq_id, "The rq_id param was not found in the server response"
+
+        response = self._wait_request_to_be_finished(user, rq_id, expected_4xx_status_code=expected_4xx_status_code)
+
+        if not download_locally:
+            return response
+
+        # get actual result URL to check that server returns 401/403 when a user tries to download prepared file
+        if expected_4xx_status_code:
+            response = self._wait_request_to_be_finished(self.admin, rq_id)
+
+        result_url = response.json().get("result_url")
+        assert result_url, "The result_url param was not found in the server response"
+
+        response = self._get_request(result_url, user)
+        self.assertEqual(response.status_code, expected_4xx_status_code or status.HTTP_200_OK)
+
+        if not expected_4xx_status_code and file_path:
+            content = BytesIO(b"".join(response.streaming_content))
+            with open(file_path, "wb") as f:
+                f.write(content.getvalue())
+
+        return response
+
+    def _get_api_path_to_export_task_backup(self, task_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/tasks/{task_id}/backup/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params)
+        return api_path
+
+    def _get_api_path_to_export_project_backup(self, project_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/projects/{project_id}/backup/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params)
+        return api_path
+
+    def _get_api_path_to_export_project_dataset(self, project_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/projects/{project_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=True)
+        return api_path
+
+    def _get_api_path_to_export_project_annotations(self, project_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/projects/{project_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=False)
+        return api_path
+
+    def _get_api_path_to_export_task_dataset(self, task_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/tasks/{task_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=True)
```

Comment:
If `query_params` is `None`, then the `save_images` parameter will never be added - presumably that's not the desired effect?

Every export requires at least the `format` parameter, though, so it seems like you could make `query_params` required.